src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -324,6 +325,22 @@ private function addWorkflowSection(ArrayNodeDefinition $rootNode)
                                         ->defaultValue([])
                                         ->prototype('scalar')->end()
                                     ->end()
+                                    ->arrayNode('dispatched_events')
+                                        ->beforeNormalization()
```

Comment:
I don't think we should handle many way to configure the same thing.
to me, the perfect config would be:

1. no configuration => all
1. an empty array => non
1. an array with few items => only theses items
---
So I would remove the case for string. And it would allow to remove special cases for `all` and `none`.